extensions/api-icon-list/src/index.tsx

```diff
@@ -1,84 +1,47 @@
-import { ActionPanel, CopyToClipboardAction, Icon, List, render } from "@raycast/api";
+import { ActionPanel, Color, CopyToClipboardAction, Icon, List } from "@raycast/api";
+import { useState } from "react";
 
-type IconItem = [string, Icon];
+function enumKeys<O extends object, K extends keyof O = keyof O>(obj: O): K[] {
+  return Object.keys(obj).filter(k => Number.isNaN(+k)) as K[];
+}
 
 export default function Command() {
-  const iconsList: IconItem[] = [
-    ["ArrowClockwise", Icon.ArrowClockwise],
-    ["ArrowRight", Icon.ArrowRight],
-    ["Binoculars", Icon.Binoculars],
-    ["Bubble", Icon.Bubble],
-    ["Calendar", Icon.Calendar],
-    ["Checkmark", Icon.Checkmark],
-    ["ChevronDown", Icon.ChevronDown],
-    ["ChevronUp", Icon.ChevronUp],
-    ["Circle", Icon.Circle],
-    ["Clipboard", Icon.Clipboard],
-    ["Clock", Icon.Clock],
-    ["Desktop", Icon.Desktop],
-    ["Document", Icon.Document],
-    ["Dot", Icon.Dot],
-    ["Download", Icon.Download],
-    ["Envelope", Icon.Envelope],
-    ["ExclamationMark", Icon.ExclamationMark],
-    ["Eye", Icon.Eye],
-    ["EyeSlash", Icon.EyeSlash],
-    ["Finder", Icon.Finder],
-    ["Gear", Icon.Gear],
-    ["Globe", Icon.Globe],
-    ["Hammer", Icon.Hammer],
-    ["LevelMeter", Icon.LevelMeter],
-    ["Link", Icon.Link],
-    ["List", Icon.List],
-    ["MagnifyingGlass", Icon.MagnifyingGlass],
-    ["MemoryChip", Icon.MemoryChip],
-    ["Message", Icon.Message],
-    ["Pencil", Icon.Pencil],
-    ["Person", Icon.Person],
-    ["Phone", Icon.Phone],
-    ["Pin", Icon.Pin],
-    ["Plus", Icon.Plus],
-    ["Sidebar", Icon.Sidebar],
-    ["SpeakerArrowDown", Icon.SpeakerArrowDown],
-    ["SpeakerArrowUp", Icon.SpeakerArrowUp],
-    ["SpeakerSlash", Icon.SpeakerSlash],
-    ["Star", Icon.Star],
-    ["Text", Icon.Text],
-    ["TextDocument", Icon.TextDocument],
-    ["QuesionMark", Icon.QuestionMark],
-    ["Terminal", Icon.Terminal],
-    ["Trash", Icon.Trash],
-    ["Upload", Icon.Upload],
-    ["Video", Icon.Video],
-    ["Window", Icon.Window],
-    ["XmarkCircle", Icon.XmarkCircle],
-  ];
+  const [color, setColor] = useState<Color>();
+  const actionPanelItems: Element[] = [];
+  for (const colorName of enumKeys(Color)) {
+    actionPanelItems.push(
+      <ActionPanel.Item
+        icon={{ source: Icon.Dot, tintColor: Color[colorName] }}
+        title={colorName}
+        key={colorName}
+        onAction={() => setColor(Color[colorName])}
+      />
+    );
+  }
 
-  return <IconsList icons={iconsList} />;
-}
+  const items: Element[] = [];
+  for (const iconName of enumKeys(Icon)) {
+    items.push(
+      <List.Item
+        title={iconName}
+        icon={{ source: Icon[iconName], tintColor: color }}
+        id={iconName}
+        key={iconName}
+        actions={
+          <ActionPanel>
+            <CopyToClipboardAction title="Copy Icon Name" content={iconName} />
```

Comment:
It could be cool to have another action to copy the code for the tinted icon. Then you can instantly apply this.

Reply:
Done in 77957bf.